src/test/java/mate/academy/booksstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,206 @@
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeEach
+    public void beforeEach(@Autowired DataSource dataSource,
+                           @Autowired WebApplicationContext context)
+            throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/add-two-books-to-books-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/add-two-categories-to-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/add-user-to-users-table.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/add-shopping-cart-with-"
+                            + "two-cart-items-to-shopping-carts-table.sql"));
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    public void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(
+                    "database/books/remove-all-books-from-books-table.sql"));
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(
+                    "database/books/remove-all-from-books-categories-table.sql"));
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(
+                    "database/books/remove-all-from-shopping-cart-table.sql"));
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(
+                    "database/books/remove-all-from-users-table.sql"));
+        }
+
+    }
+
+    @Test
+    @WithMockUser()
+    @DisplayName("Get all cart items in shopping cart")
+    public void getShoppingCart_givenUserId_returnShoppingCartDto() throws Exception {
+        String token = jwtUtil.generateToken("admin@example.com");
+        MvcResult result = mockMvc.perform(get("/api/cart")
+                        .header("Authorization", "Bearer " + token))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(result
+                .getResponse()
+                .getContentAsString(), ShoppingCartDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(2, actual.getCartItems().size());
```

Comment:
You dont need objectMapper to test your controller. 
It is awful solution. 
Consider using andExpect() + jsonPath()

```suggestion
```